### PR TITLE
server/etcdmain: add configurable cipher list to gRPC proxy listener

### PR DESCRIPTION
**Pull request by dusk125**
This PR essentially reopens #13362 cherrypicked onto main and with the requested test.

### CODE REVIEW

**Comment by tjungblu** [comment]
hah now I can :police_officer: you have to `commit -s` and sign the commit

File: tests/e2e/etcd_config_test.go [hunk 1]
```diff
@@ -302,6 +302,44 @@ func TestGrpcproxyAndCommonName(t *testing.T) {
...
+		"start",
+		"--listen-cipher-suites", "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256",
+	}
+
+	argsWithoutCipherSuites := []string{
+		binDir + "/etcd",
+		"grpc-proxy",
+		"start",
+		"--listen-cipher-suites", "",
+	}
```
Comment (ahrtr) [inline tests/e2e/etcd_config_test.go:320]: Please use golang [subtest](https://go.dev/blog/subtests) to update the test case, which should can improve the readability.

Comment (dusk125) [reply tests/e2e/etcd_config_test.go]: Thanks! This is fixed in the most recent update.

File: tests/e2e/etcd_config_test.go [hunk 2]
```diff
@@ -302,6 +302,44 @@ func TestGrpcproxyAndCommonName(t *testing.T) {
...
+				"start",
+				"--listen-cipher-suites", "",
+			},
+		},
+	}
+
+	for _, test := range cases {
+		pw, err := e2e.SpawnCmd(test.args, nil)
+		if err != nil {
+			t.Fatal(fmt.Errorf("test %v failed to start: %v", test.name, err))
```
Comment (ahrtr) [inline tests/e2e/etcd_config_test.go:335]: Please use `t.Fatalf `instead in this case ( no need to use `fmt.Errorf`)

File: tests/e2e/etcd_config_test.go [hunk 3]
```diff
@@ -302,6 +302,44 @@ func TestGrpcproxyAndCommonName(t *testing.T) {
...
+		}
+		if err = pw.Stop(); err != nil {
+			t.Fatal(fmt.Errorf("test %v failed to stop: %v", test.name, err))
```
Comment (ahrtr) [inline tests/e2e/etcd_config_test.go:338]: same as above

File: tests/e2e/etcd_config_test.go [hunk 4]
```diff
@@ -302,6 +302,44 @@ func TestGrpcproxyAndCommonName(t *testing.T) {
...
+				e2e.BinDir + "/etcd",
+				"grpc-proxy",
+				"start",
+				"--listen-cipher-suites", "",
+			},
+		},
+	}
+
+	for _, test := range cases {
+		pw, err := e2e.SpawnCmd(test.args, nil)
```
Comment (ahrtr) [inline tests/e2e/etcd_config_test.go:333]: Please get the test wrappted in [t.Run](https://github.com/golang/go/blob/346b18ee9d15410ab08dd583787c64dbed0666d2/src/testing/testing.go#L1273). 

Example: [version_test.go#L75](https://github.com/etcd-io/etcd/blob/b2726c4da396d66edc744248befb876d9cfdd036/api/version/version_test.go#L75)

**Comment by ahrtr** [comment]
Please fix the format error, try to run `gofmt -w the_file_you_changed`

**Comment by ahrtr** [comment]
Please also add an item into the [changelog-3.6](https://github.com/etcd-io/etcd/blob/main/CHANGELOG/CHANGELOG-3.6.md)

File: CHANGELOG/CHANGELOG-3.6.md [hunk 5]
```diff
@@ -81,6 +81,7 @@ See [code changes](https://github.com/etcd-io/etcd/compare/v3.5.0...v3.6.0).
 ### etcd grpc-proxy
 
 - Add [`etcd grpc-proxy start --endpoints-auto-sync-interval`](https://github.com/etcd-io/etcd/pull/14354) flag to enable and configure interval of auto sync of endpoints with server.
+- Add [listen-cipher-suites](https://github.com/etcd-io/etcd/pull/14308) flag.
```
Comment (ahrtr) [inline CHANGELOG/CHANGELOG-3.6.md:84]:
```suggestion
- Add [`etcd grpc-proxy start --listen-cipher-suites`](https://github.com/etcd-io/etcd/pull/14308) flag to support adding configurable cipher list.
```

**Review by ahrtr — APPROVED** [review]
LGTM

Thank you @dusk125

**Review by spzala — APPROVED** [review]
lgtm, these changes were long awaited. Thanks @dusk125

**Comment by tjungblu** [comment]
@dusk125 I think we also need this in 3.5 and 3.4 respectively since the report dates back into openshift 4.8

**Pull request merged**